### DISCUSSION TITLE
How to programatically generate poster from .GLTF?

### DISCUSSION THREAD

**Discussion by ronpadz** [google/model-viewer#1876] [Q&A]
While model-viewer provides excellent integration of posters to address various loading scenarios, I'm struggling to identify a straightforward means to auto-generate a poster from .gltf. This is explicitly not part of the .GLtf standard. While it would be a nice-to-have capability of model-viewer to somehow generate the poster, I'm really just looking for a method available today through other means -- preferably a passive method that can run on the server that a .GLTF is uploaded to and does not require a visible window to open. Any ideas?

**Comment by jsantell** [comment]
I'd imagine loading the model via model-viewer in a headless browser (via xvfb, etc.) and using the element's `toDataURL(type, encoderOptions)` method would achieve that

**Comment by elalish** [comment]
@ronpadz In fact I just added seamless poster transitions (it's in master now, but not yet released), along with an improved method for generating them: https://github.com/GoogleWebComponents/model-viewer/pull/890/files#diff-da247232246d3b6eb3f40f77a6473c81R339. You'll want `toBlob` with `idealAspect = true`. Then follow @jsantell directions. You definitely want to capture the poster using model-viewer, as that is the only way to get the camera set up just right to get seamless transitions.

**Comment by cdata** [comment]
@ronpadz echoing what @jsantell and @elalish have said, we have some methods on `<model-viewer>` today to help with this, and I would particularly second @elalish 's recommendation of `toBlob({ idealAspect: true })`.

Of course, if you use headless Chrome you can generate a screenshot of anything (this is how we produce screenshots for our fidelity tests, for example). You can see how we wait for models to render and then capture a screenshot w/ headless Chrome via [puppeteer](https://developers.google.com/web/tools/puppeteer) here:

https://github.com/GoogleWebComponents/model-viewer/blob/3fb2fc30cc80ce66032c6b8da91abc10736b444f/src/test/fidelity/artifact-creator.ts#L176-L237

**Comment by ronpadz** [comment]
@cdata @jsantell @elalish This looks to be the info I'm looking for. Can't wait to give it a try. Muchas gracias!

**Comment by osagrovVi3** [comment]
So one thing I noticed so far is that the poster image works well when I do a full browser screenshot, and crop the image width to be a square (or standing) image. Took a little guesswork. Could be nice to have this information in the documentation.

**Reply by elalish** [reply]
It's documented here: https://modelviewer.dev/docs/index.html#entrydocs-loading-methods-toBlob

Also you can generate them with our [editor](https://modelviewer.dev/editor/).

**Reply by osagrovVi3** [reply]
Thank you, this is really helpful.
